### ISSUE TITLE
[FEATURE REQUEST]: copy table as table

### ISSUE THREAD

**Issue by DutchPete** [marktext/marktext#1329]
### Describe your feature request

 When I copy a table from Excel and paste it into MT while MT is in *Source Code Mode*, I would like it to be copied with the borders, so that it looks like a proper table in the *Editor* mode.

### What problem does this feature solve? [optional]

 Currently pasting while MT is in *Source Code Mode* does not paste the borders, so in *Editor* mode it does not look like a table but like a pile of text.

**Comment by fxha** [comment]
I think this is a duplicate of #1271 because the content is not recognized as table. It's just inserted as text like the following, right?

```
a	c	v
a	d	d
d	d	d
```

**Comment by DutchPete** [comment]
You could look at it as 2 aspects of the same issue. What it boils down to is that a table that is copied from a spreadsheet, is pasted as:
* either an image, as I explained in #1271, or
* it is pasted as a jumbled pile of text, as I explained above (#1329).

For some people pasting as an image is fine, for others, like me, it is not. Hence my request for an enhancement.

**Comment by fxha** [comment]
> For some people pasting as an image is fine

It's not expected to be an image. I think #1271 should also fix this issue because Mark Text don't recognize tables pasted from LibreOffice as table but plain text.

**Comment by fxha** [comment]
Blocked by #1271.

**Comment by fxha** [comment]
@DutchPete We currently don't support pasting tables, images or other special stuff in source-code mode.

To support this, CodeMirror requires custom paste handlers to detect and parse HTML. This could be done via `paste` and `beforeChange` event but would require a lot of code that is used inside muya.

**Comment by DutchPete** [comment]
@fxha : thanks for your feedback. I understand the difficulty, which means it won't be incorporated in 0.15.0.

In a comment from @Jocs above it says *"added this to TO-DO in 0.16.0"*.
Is that still on? In other words, may I assume it will be in version 0.16.0 ?

**Comment by fxha** [comment]
It's currently delayed because it's just the source-code code and the main priority is preview mode. We also need to discuss how to implement this without duplicating a lot of code.
